mathesar/tests/views/api/test_table_api.py

```diff
@@ -120,6 +121,34 @@ def test_table_detail(create_table, client):
     check_table_response(response_table, table, table_name)
 
 
+def test_table_type_suggestion(client, test_db_name):
+    table_name = 'Type Inference Table'
+    file = 'mathesar/tests/data/type_inference.csv'
+    schema = 'type_inference'
+    EXPECTED_TYPES = {
+        'col_1': 'numeric',
+        'col_2': 'boolean',
+        'col_3': 'boolean',
+        'col_4': 'string',
+        'col_5': 'string',
+        'col_6': 'numeric'
+    }
+
+    with open(file, 'rb') as csv_file:
```

Comment:
We shouldn't be using the legacy function in new code since it's scheduled to be deleted.

Reply:
Do we just want to use the table creation API instead?

Reply:
Yes.